src/main/java/org/truffleruby/core/regexp/RegexpOptions.java

```diff
@@ -153,7 +153,7 @@ public Encoding setup(RubyContext runtime) {
             return EUCJPEncoding.INSTANCE;
         } else if (explicitKCode == KCode.SJIS) {
             setFixed(true);
-            return Layouts.ENCODING.getEncoding(runtime.getEncodingManager().getRubyEncoding(WINDOWS31J));
+            return EncodingManager.getEncoding(WINDOWS31J);
```

Comment:
I was just looking at that and I think that's the case. The class isn't normally loaded and the first time it is it populates its lookup tables. I assume in general the branch isn't taken frequently. For TruffleRuby is unlikely to be taken ever, since we don't support Windows.

Is there really a problem with looking it up by name?